pyface/i_layered_panel.py

```diff
@@ -0,0 +1,99 @@
+""" Interace and mixins for layered panels.
+
+A layered panel contains one or more named layers, with only one layer
+visible at any one time (think of a 'tab' control minus the tabs!).
+"""
+
+from traits.api import Any, Dict, HasTraits, Interface, Str
+
+
+class ILayeredPanel(Interface):
+    """ A Layered panel.
+
+    A layered panel contains one or more named layers, with only one layer
+    visible at any one time (think of a 'tab' control minus the tabs!).  Each
+    layer is a toolkit-specific control.
+
+    """
+
+    # "ILayeredPanel' interface --------------------------------------------
+
+    # The toolkit-specific control of the currently displayed layer.
+    current_layer = Any()
+
+    # The name of the currently displayed layer.
+    current_layer_name = Str()
+
+    # ------------------------------------------------------------------------
+    # 'ILayeredPanel' interface.
+    # ------------------------------------------------------------------------
+
+    def add_layer(self, name, layer):
+        """ Adds a layer with the specified name.
+
+        All layers are hidden when they are added.  Use 'show_layer' to make a
+        layer visible.
+
+        """
+
+    def show_layer(self, name):
+        """ Shows the layer with the specified name. """
+
+    def has_layer(self, name):
+        """ Does the panel contain a layer with the specified name? """
+
+
+class MLayeredPanel(HasTraits):
+    """ A Layered panel mixin.
+
+    A layered panel contains one or more named layers, with only one layer
+    visible at any one time (think of a 'tab' control minus the tabs!).  Each
+    layer is a toolkit-specific control.
+    """
+
+    # "ILayeredPanel' interface --------------------------------------------
+
+    # The toolkit-specific control of the currently displayed layer.
+    current_layer = Any()
+
+    # The name of the currently displayed layer.
+    current_layer_name = Str()
+
+    # Private traits -------------------------------------------------------
+
+    # The a map of layer names to toolkit controls in the panel.
+    _layers = Dict(Str)
+
+    # ------------------------------------------------------------------------
+    # 'ILayeredPanel' interface.
+    # ------------------------------------------------------------------------
+
+    def has_layer(self, name):
+        """ Does the panel contain a layer with the specified name? """
+        return name in self._layers
+
+    # ------------------------------------------------------------------------
+    # 'object' interface.
+    # ------------------------------------------------------------------------
+
+    def __init__(self, parent=None, **traits):
+        """ Creates a new LayeredPanel. """
+
+        create = traits.pop("create", True)
+
+        # Base class constructor.
+        super().__init__(parent=parent, **traits)
+
+        if create:
+            # Create the toolkit-specific control that represents the widget.
+            self.create()
```

Comment:
what about the warnings related to not automatically calling `create`?

Reply:
Thanks.